tools/autograd/gen_python_functions.py

```diff
@@ -186,7 +187,7 @@ def gen_py_nn_functions(out, declarations, template_path):
 
     def should_bind(declaration):
         return (should_generate_python_binding(declaration) and
-                declaration['mode'] == 'NN')
+                (declaration['mode'] == 'NN' or declaration.get('python_module') == 'nn'))
```

Comment:
When can we get rid of the `declaration['mode']` test?

Reply:
Good question; the plan is subject to change but roughly:
1) Implement native_wrappers for all nn.yaml functions
2) Make actual nn.yaml functions (i.e. not the native wrappers) an "implementation" detail in that they don't end up in Declarations.yaml, then we don't need declaration['mode'] at all -- everything will be native.  We may still have something similar for differentiating 'TH' vs 'NN' implementation details, but that shouldn't leak anywhere.